airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java

```diff
@@ -391,9 +412,56 @@ private void putLongArray(final ObjectNode node, final String columnName, final
     node.set(columnName, arrayNode);
   }
 
+  @Override
+  protected void putDate(ObjectNode node, String columnName, ResultSet resultSet, int index) throws SQLException {
+    String strValue = resultSet.getString(index);
+    if (INFINITY_STRING.equals(strValue)) {
+      node.put(columnName, PLUS_INFINITY_STRING);
+    } else if (MINUS_INFINITY_STRING.equals(strValue)) {
+      node.put(columnName, MINUS_INFINITY_STRING);
+    } else {
+      // even though the super just does a toString, I find it a lot cleaner to call it anyways, in case
+      // that implementation changes
+      super.putDate(node, columnName, resultSet, index);
+    }
+  }
+
+  @Override
+  protected void putTime(ObjectNode node, String columnName, ResultSet resultSet, int index) throws SQLException {
+    super.putTime(node, columnName, resultSet, index);
+  }
+
   @Override
   protected void putTimestamp(final ObjectNode node, final String columnName, final ResultSet resultSet, final int index) throws SQLException {
-    node.put(columnName, DateTimeConverter.convertToTimestamp(resultSet.getTimestamp(index)));
+    String strValue = resultSet.getString(index);
+    if (INFINITY_STRING.equals(strValue)) {
+      node.put(columnName, PLUS_INFINITY_STRING);
```

Comment:
what actually happens on destination when receiving "infinity" value for a timestamp?